File: Example/UnitTests/UICorePluginTests.m
#import <Clappr/Clappr.h>

SPEC_BEGIN(UICorePlugin)

describe(@"UICorePlugin", ^{

    CLPCore *core = [[CLPCore alloc] initWithSources:@[]];;

    describe(@"instantiation", ^{

        it(@"should have a designated initializer receiving the core instance", ^{
            CLPUICorePlugin *plugin = [[CLPUICorePlugin alloc] initWithCore:core];
            [[plugin.core should] equal:core];
        });

        it(@"should raise an exception for its default initializer", ^{
            [[theBlock(^{
                [CLPUICorePlugin new];
            }) should] raiseWithName:NSInternalInconsistencyException];
        });

        it(@"should have its view as a child of core's view", ^{
            CLPUICorePlugin *plugin = [[CLPUICorePlugin alloc] initWithCore:core];
            [[plugin.superview should] equal:core];
        });
    });

});

SPEC_END

File: Example/Tests/UICorePluginTests.swift
import Quick
import Nimble
import Clappr

class UICorePluginTests: QuickSpec {
    
    override func spec() {
        describe("Initialization") {
            let core = Core(sources: [])
            
            it("Should have a initializer that receives a core instance") {
                let plugin = UICorePlugin(core: core)
                expect(plugin.core) == core
            }
            
            it("Should have it's view as a child of core view") {
                let plugin = UICorePlugin(core: core)
                expect(plugin.superview) == core
            }
        }
    }
}